src/LiveComponent/assets/test/controller/render.test.ts

```diff
@@ -186,6 +186,22 @@ describe('LiveController rendering Tests', () => {
         expect(test.element.innerHTML).toContain('I should not be removed');
     });
 
+    it('throws an error if the template does not starts with a single live controller root', async () => {
+        const test = await createTest(
+            {},
+            (data: any) => `
+            <div data-controller="some-other-not-live-controller">
+                <div ${initComponent(data)}></div>
+            </div>
+        `
+        );
+
+        test.expectsAjaxCall('get').expectSentData(test.initialData).init();
+
+        // @fixme how to assert this?
+        await waitFor(() => expect(test.component.render()).rejects.toThrow('I should fail'));
```

Comment:
I don't know how to do this

Reply:
Hmm. `test.component.render()` returns a `Promise`... and the `Promise, thanks to your exception, should get rejected. I've not done this before, but did you try `rejects.toEqual()`? https://jestjs.io/docs/tutorial-async#rejects

Btw, I like this error. If we get hung up on the test, I don't think that's too important.

Reply:
Thanks, I tried a lot of variations, but none of them worked. Maybe a returned promise is not connected with that part of the code, that's the backend response which already resolved by then. Anyway I removed the test.